### ISSUE TITLE
Allow runtime choice of TLS 1.2 and TLS 1.3

### ISSUE THREAD

**Issue by hanno-arm** [Mbed-TLS/mbedtls#4823]
Adjust TLS 1.3 prototype implementation so that the choice of TLS 1.2 and TLS 1.3 can be negotiated at runtime.

Also, study relation between `mbedtls_ssl_session.minor_ver` and `mbedtls_ssl_context.minor_ver` and whether one of them can be removed. See https://github.com/ARMmbed/mbedtls/pull/4806#discussion_r680510002.

Depends on: https://github.com/ARMmbed/mbedtls/issues/4224

**Comment by hanno-arm** [comment]
Thoughts: Ideally, would consolidate reading and writing of `ClientHello` routines, and dispatch to TLS 1.2 vs. TLS 1.3 state machine afterwards.

**Comment by jethrogb** [comment]
Is there a separate ticket for server support? I haven't found it yet
